test/e2e/util/nfs_cluster.go

```diff
@@ -103,6 +105,18 @@ func NFSServerTest(t *testing.T, ctx *framework.TestCtx) {
 		t.Fatal(err)
 	}
 
+	// Check if a ServiceMonitor was created.
+	// ServiceMonitor is only created when the ServiceMonitor CRD is known in
+	// the cluster.
+	serviceMonitor := &monitoringv1.ServiceMonitor{}
+	smNSName := types.NamespacedName{
+		Name:      fmt.Sprintf("%s-%s", testNFSServer.Name, "metrics"),
+		Namespace: defaultNS,
+	}
+	if err := f.Client.Get(goctx.TODO(), smNSName, serviceMonitor); err != nil {
+		t.Error("failed to get NFS metrics ServiceMonitor", err)
+	}
```

Comment:
This is an example of using a dynamic client for third party custom resource, instead of importing a generated clientset. We can do the same for the consumers of our operator's CRDs and get rid of the clientset generation part.

Reply:
Interesting... that would work in the places where we've used the clientset.